tests/bson/bson-toPHP-015.phpt

```diff
@@ -0,0 +1,140 @@
+--TEST--
+MongoDB\BSON\toPHP(): Assigning a type map value other than default disables persistable behaviour
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+class MyDocument implements MongoDB\BSON\Persistable
+{
+    public $data;
+
+    public function __construct()
+    {
+        $this->data = array(
+            'list' => array(1, 2, 3),
+            'map' => (object) array('foo' => 'bar'),
+        );
+    }
+
+    #[\ReturnTypeWillChange]
+    public function bsonSerialize()
+    {
+        return $this->data;
+    }
+
+    public function bsonUnserialize(array $data): void
+    {
+        foreach (array('list', 'map') as $key) {
+            if (isset($data[$key])) {
+                $this->data[$key] = $data[$key];
+            }
+        }
+    }
+}
+
+$tests = [
+    'root document' => [
+        new MyDocument,
+        ['root' => 'object'],
```

Comment:
Per my earlier comment, you might as well add tests for all possible values -- both the string literals and a custom class name.

If we don't already have a test for the `__pclass` prioritization I mentioned above, it'd be good to add that as well. That would likely require making a fake `__pclass` value of a class that (a) doesn't exist and (b) exists but doesn't implement the interface, and demonstrating that another class name in the type map can take priority. The class that takes priority would need to implement Unserializable of course.

Reply:
I added tests for other values and found a small behavioural change that I was able to undo. 👍